### PR TITLE
add support for custom query parameters

### LINKED ISSUE #509: allow passing of custom key/value parameters applied to all host URL
it can be useful to allow the specification of custom url parameters. the config setting would look like:

`config :custom_parameters, :validate => Hash`

examples of usage would be

```
output {
  elasticsearch {
    hosts => [ "host1.lan:9200" ]
    custom_parameters => {
      "info" => "logstash"
      "number => 2
    }
    path => "/es"
  }
}
```

this would make logstash post events to: `http://host1.lan:9200/es/_bulk?info=logstash&number=2`

an implementation of this must be aware that other parameters can be set (e.g. scripting or routing parameters) and these can clash.

### PR DESCRIPTION
**Pull request by jsvd**
fixes #509

### CODE REVIEW

**Comment by jsvd** [comment]
@andrewvc if you could take a look at this, it would be great. I'm not super sold on the name of the config (`parameters`). it's kind of a query_string, but in hash format.. query_hash?

File: lib/logstash/outputs/elasticsearch.rb [hunk 1]
```diff
@@ -103,6 +103,11 @@ class LogStash::Outputs::ElasticSearch < LogStash::Outputs::Base
   # not also set this field. That will raise an error at startup
   config :path, :validate => :string
 
+  # Pass a set of key value pairs as the URL query string. This query string is added
+  # to every host listed in the 'hosts' configuration. If this list contains urls which
+  # already contain a query string, this one will be appended to the existing one.
```
Comment (andrewvc) [inline lib/logstash/outputs/elasticsearch.rb]: Wait, why would someone put a full URL in here? This sentence "If this list contains urls which
 +# already contain a query string, this one will be appended to the existing one." is very confusing to me.

Comment (jsvd) [inline lib/logstash/outputs/elasticsearch.rb]: oh, I meant, if you paste a url in `hosts` like `hosts => [ "http://localhost?random=thing" ]`, the specified query parameters are added to `random=thing`

**Comment by jsvd** [comment]
@andrewvc I've updated the documentation phrasing. I'm still not super happy with the english, any suggestion is welcome

**Comment by andrewvc** [comment]
@jsvd the changes you made look good. Your english is great, just needed a second pass :)

**Pull request closed without merging**